lib/checknullpointer.cpp

```diff
@@ -279,47 +280,72 @@ static bool isNullablePointer(const Token* tok)
     return false;
 }
 
-void CheckNullPointer::nullPointerByDeRefAndChec()
+static bool isPointerUnevaluated(const Token *tok)
+{
+    if (isUnevaluated(tok))
+        return true;
+
+    while (tok) {
+        if (isUnevaluated(tok->previous()))
```

Comment:
could you clarify what will `tok->previous()` be. if there is parenthesis like `...)tok..` or something then you want to return true?

Reply:
I want to make sure that the following code does not trigger a warning.
```
void f(type* p) {
   x(sizeof ( (p[0])));
    if (!p)
        ;
}
```
I need to check whether the variable `p` is unevaluated so I used a similar method found in `checkuninitvar.cpp` (see method `CheckUninitVar::checkExpr`).